tendrl/node_agent/provisioner/ceph/plugins/ceph_installer.py

```diff
@@ -0,0 +1,254 @@
+            except (TypeError, ValueError, UnicodeError) as e:
+                raise Exception(
+                    'Server response was not valid JSON: %r' % e)
+            return res_data['identifier']
+        else:
+            return None
+
+    def install_osd(self, osds):
+        url = 'http://localhost:%s/api/osd/install' % \
+            self._CEPH_INSTALLER_API_PORT
+
+        # Sample payload for osd install API is
+        # Content-Type: application/json
+        # {
+        #     "hosts": ["osd1.example.com", "osd2.example.com"],
+        #     "redhat_storage": false,
+        #     "redhat_use_cdn": true,
+        #     "verbose": false,
+        # }
+
+        data = {
+            "hosts": osds,
+            "redhat_storage": False,
+            "redhat_use_cdn": True,
+            "verbose": False,
+        }
+        resp = self.http.request(
+            self._MPOST,
+            url,
+            body=json.dumps(data),
+            headers={'Content-Type': 'application/json'})
+        if resp.status == 200:
+            try:
+                res_data = json.loads(resp.data.decode('utf-8'))
+            except (TypeError, ValueError, UnicodeError) as e:
+                raise Exception(
+                   'Server response was not valid JSON: %r' % e)
+            return res_data['identifier']
+        else:
+            return None
+
+    def configure_mon(
+        self,
+        host,
+        cluster_id,
+        cluster_name,
+        cluster_network,
+        public_network,
+        mons
+    ):
+        url = 'http://localhost:%s/api/mon/configure' % \
+            self._CEPH_INSTALLER_API_PORT
+
+        # Sample payload for osd install API is
+        # Content-Type: application/json
+        # {
+        #     "calamari": false,
+        #     "conf": {"global": {"auth supported": "cephx"}},
+        #     "host": "mon1.example.com",
+        #     "interface": "eth0",
+        #     "fsid": "deedcb4c-a67a-4997-93a6-92149ad2622a",
+        #     "monitor_secret": "AQA7P8dWAAAAABAAH/tbiZQn/40Z8pr959UmEA==",
+        #     "cluster_name": "my-ceph-cluster",
+        #     "cluster_network": "192.0.2.0/24",
+        #     "public_network": "198.51.100.0/24",
+        #     "monitors": [{"host": "mon0.host", "interface": "eth0"}],
+        #     "redhat_storage": false,
+        #     "verbose": false,
+        # }
+        #
+        # Note: monitors array is optional. This is only optional when no
+        # other monitors currently exist in the cluster. If you are configuring
+        # a mon for an existing cluster, provide a list of objects representing
+        # the monitor host and its interface or address.
+
+        data = {
+            "calamari": False,
+            "conf": {"global": {"auth supported": "cephx"}},
+            "host": host,
+            "interface": "eth0",
+            "fsid": cluster_id,
+            "monitor_secret": "AQA7P8dWAAAAABAAH/tbiZQn/40Z8pr959UmEA==",
```

Comment:
we can have this configurable if required. earlier also we had this hardcoded...